spotipy/client.py

```diff
@@ -240,6 +240,10 @@ def _internal_call(self, method, url, payload, params):
                 msg = response.json()["error"]["message"]
             except (ValueError, KeyError):
                 msg = "error"
+            try:
+                reason = response.json()["error"]["reason"]
```

Comment:
we could re-use the same `response.json()` as above, then check `if "error" in json_response`. And from that `error`, populate `msg` and `reason` accordingly

Reply:
That looks good!